golem-worker-service-base/src/service/worker/default.rs

```diff
@@ -735,6 +745,63 @@ where
         Ok(metadata)
     }
 
+    async fn get_metadatas(
+        &self,
+        template_id: &TemplateId,
+        filter: Option<WorkerFilter>,
+        cursor: u64,
+        count: u64,
+        precise: bool,
+        _auth_ctx: &AuthCtx,
+    ) -> WorkerResult<(Option<u64>, Vec<WorkerMetadata>)> {
+        let result = self.execute_with_random_client(
+            &(template_id.clone(), filter.clone(), cursor, count, precise),
+            |worker_executor_client, (template_id, filter, cursor, count, precise)| {
+                Box::pin(async move {
+                    let template_id: golem_api_grpc::proto::golem::template::TemplateId =
+                        template_id.clone().into();
+                    let response = worker_executor_client.get_worker_metadatas(
+                        golem_api_grpc::proto::golem::workerexecutor::GetWorkerMetadatasRequest {
+                            template_id: Some(template_id),
+                            filter: filter.clone().map(|f| f.into()),
+                            cursor: *cursor,
+                            count: *count,
+                            precise: *precise,
+                        }
+                    ).await.map_err(|err| {
+                        GolemError::RuntimeError(GolemErrorRuntimeError {
+                            details: err.to_string(),
+                        })
+                    })?;
+                    match response.into_inner() {
+                        workerexecutor::GetWorkerMetadatasResponse {
+                            result:
+                            Some(workerexecutor::get_worker_metadatas_response::Result::Success(workerexecutor::GetWorkerMetadatasSuccessResponse {
+                                workers, cursor })),
+                        } => {
+                            let cursor: Option<u64> = if cursor == 0 { None } else { Some(cursor) };
+                            let workers = workers.iter().map(| w| w.clone().try_into()).collect::<Result<Vec<_>, _>>().map_err(|_| GolemError::Unknown(GolemErrorUnknown {
```

Comment:
Using the `Into`/`From`/`TryInto`/`TryFrom` instances instead of manual conversion functions makes a lot of things cleaner and automatic so I'm fully supporting it, and already got rid of most of the manual conversion codes in our codebase.

If readability suffers that can easily be fixed by some more explicit type annotations.

(Actually I have similar feelings that you have, but for type inference in general :)